sql/core/src/main/scala/org/apache/spark/sql/execution/SparkSqlParser.scala

```diff
@@ -386,7 +386,7 @@ class SparkSqlAstBuilder(conf: SQLConf) extends AstBuilder {
         "LOCATION and 'path' in OPTIONS are both used to indicate the custom table path, " +
           "you can only specify one of them.", ctx)
     }
-    val customLocation = storage.locationUri.orElse(location)
+    val customLocation = storage.locationUri.orElse(location.map(CatalogUtils.stringToURI(_)))
 
```

Comment:
Also, it seems we have a similar util with `CatalogUtils.stringToURI` in `org.apache.spark.util.Utils.resolveURI`. Could we consolidate them? I did not replace it when I identified this because some existing tests in `org.apache.spark.util.UtilsSuite` were broken but I guess it would be fine if there is a coherent reason. These broken cases might be bugs.

@windpiger could you double check my comments and open a followup if I was not wrong?